src/env/as_env.rs

```diff
@@ -152,3 +153,54 @@ pub fn assembly_script_console_log(
     env.get_interface().generate_event(message)?;
     Ok(())
 }
+
+/// Assembly script builtin `trace()`.
+#[named]
+pub fn assembly_script_trace(
+    mut ctx: FunctionEnvMut<ASEnv>,
+    message: StringPtr,
+    n: i32,
+    a0: f64,
+    a1: f64,
+    a2: f64,
+    a3: f64,
+    a4: f64,
+) -> ABIResult<()> {
```

Comment:
Would an input array for trace not be compatible with AS? This behaviour for the trace builtin seems very awkward

Reply:
If not you need to disable the clippy warning

Reply:
Here is the prototype for the trace function in AS: 

function trace(message: string, n?: i32, a0?: f64, a1?: f64, a2?: f64, a3?: f64, a4?: f64): void

and the corresponding prototype in WASM:

trace(i32, i32, f64, f64, f64, f64, f64)

So I'm going to disable the clippy warning...

Reply:
done